### PR TITLE
CICD Docs (Github/Gitlab)

### LINKED ISSUE #405: Add documentation page for CI integration
We currently provide an intro to using the GitHub Action here: https://github.com/StyraInc/regal#github-actions

I think it would be good to break that out into a separate page on the topic of CICD integration, where we could include GitLab, and possibly other build systems.

### PR DESCRIPTION
**Pull request by Parsifal-M**
Hey Team! :wave: 

This closes #405 and adds a separate CICD doc :smiley: 

I moved the Github one (as-is) from the README and added what we use for Regal in Gitlab CICD (which works for us).

LMK in terms of wording or structure if its okay and if not I will update it :+1: 

Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,30 +152,9 @@ Documentation:	https://docs.styra.com/regal/rules/style/prefer-snake-case
 > will likely generate a lot of violations. You can do this by passing the `--disable-category style` flag to
 > `regal lint`.
 
-### GitHub Actions
+### Using Regal in CI/CD Workflows
```
Comment (anderseknert) [inline README.md:155]: "Using Regal in your build pipeline" sounds a little less techno-babbly, perhaps?

File: README.md [hunk 2]
```diff
@@ -152,30 +152,9 @@ Documentation:	https://docs.styra.com/regal/rules/style/prefer-snake-case
...
-      with:
-        # For production workflows, use a specific version, like v0.16.0
-        version: latest
-
-    - name: Lint
-      run: regal lint --format=github ./policy
-```
-
-Please see [`setup-regal`](https://github.com/StyraInc/setup-regal) for more information.
+Kindly refer to the [CICD](/docs/cicd.md) documentation for more information on how to integrate Regal into your CI/CD workflows.
```
Comment (anderseknert) [inline README.md:157]: A little more context here would be helpful. Perhaps something like:

> To ensure Regal's rules are enforced consistently in your project or organization, we've made it easy to run Regal as part of your builds. See the docs on "Using Regal in your build pipeline" to learn more about how to set up Regal to lint your policies on every commit or pull request.

File: docs/cicd.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+on:
+  pull_request:
+jobs:
+  lint-rego:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: StyraInc/setup-regal@v1
+      with:
+        # For production workflows, use a specific version, like v0.16.0
```
Comment (anderseknert) [inline docs/cicd.md:23]: While it's going to be outdated soon anyway, let's bump this to v0.22.0 while we're at it.

File: docs/cicd.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## GitLab CICD
+
+To use Regal in GitLab CI/CD, you could for example use the following stage in your `.gitlab-ci.yml`:
+
+```yaml
+regal_lint_policies:
+  stage: regal-lint
+  image:
+    # For production workflows, use a specific version, like v0.16.0
```
Comment (anderseknert) [inline docs/cicd.md:40]: Same here, bump version.

File: docs/cicd.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    - name: Lint
+      run: regal lint --format=github ./policy
+```
+
+Please see [`setup-regal`](https://github.com/StyraInc/setup-regal) for more information.
+
+## GitLab CICD
+
+To use Regal in GitLab CI/CD, you could for example use the following stage in your `.gitlab-ci.yml`:
```
Comment (anderseknert) [inline docs/cicd.md:34]: It's probably obvious to someone who uses GitLab, but the GH example above specifically mentions running Regal on PRs, while this sentence says nothing about when/how this is triggered. Could you add that? 🙂

**Review by anderseknert — APPROVED** [review]
LGTM! Just a few thoughts, but nothing blocking.

**Review by anderseknert — APPROVED** [review]
Thanks! 👏

**Comment by Parsifal-M** [comment]
Think it should be okay now :sweat_smile: the markdown linter is a fussy one :laughing:

**Pull request merged**